# HG changeset patch
# User Punarbasu Purkayastha <ppurka@gmail.com>
# Date 1381559295 -28800
# Node ID 3c901109788d906f8550daf049525dd4919c843c
# Parent  f102f013e8cadd7a8ad40bffb8bbdaa3c6609e07
Add _sympy_ method to real_mpfr

This patch is from http://ask.sagemath.org/question/2835/error-while-integrate-using-algorithmsympy-bug
provided by the user AndreWin

diff --git a/sage/rings/real_mpfr.pyx b/sage/rings/real_mpfr.pyx
--- a/sage/rings/real_mpfr.pyx
+++ b/sage/rings/real_mpfr.pyx
@@ -2137,6 +2137,22 @@
         mpfr_sub(x.value, self.value, (<RealNumber>right).value, (<RealField_class> self._parent).rnd)
         return x
 
+    def _sympy_(self):
+        """
+        Return a sympy object of ``self``.
+
+        EXAMPLES:
+
+        An indirect doctest to check this (see :trac:`14915`)::
+
+            sage: x,y = var('x, y')
+            sage: integrate(y, y, 0.5, 8*log(x), algorithm='sympy')
+            32*log(x)^2 - 0.125000000000000
+
+        """
+        import sympy
+        return sympy.simplify(float(self))
+
     cpdef RingElement _mul_(self, RingElement right):
         """
         Multiply two real numbers with the same parent.
